### ISSUE TITLE
Add Comprehensive Docstrings to Codebase

### ISSUE THREAD

**Issue by AmrElsayyad** [abhi9720/BankingPortal-API#28]
**Is your feature request related to a problem? Please describe.**
Currently, the codebase lacks comprehensive docstrings, making it difficult for developers to understand the purpose and functionality of various methods and classes. This can lead to confusion and increased time spent on code review and debugging.

**Describe the solution you'd like**
I propose adding detailed docstrings to all methods and classes in the codebase. These docstrings should follow a consistent format and provide clear explanations of the parameters, return values, and any exceptions raised.

**Describe alternatives you've considered**
Creating a separate documentation file is an alternative solution, but it might not be as effective as embedding docstrings directly within the code. Docstrings are more accessible to developers, making it easier for them to understand and maintain the code.

**Additional context**
Adding comprehensive docstrings will improve code readability and maintainability, making it easier for both current and future developers to work with the codebase.

**Comment by abhi9720** [comment]
Hi @AmrElsayyad 

That would be very best for new people and also helps us In debugging 

We can start using the javadoc plugin in pom.xml file

What do you think about this ?

**Comment by AmrElsayyad** [comment]
> Hi @AmrElsayyad 
> 
> That would be very best for new people and also helps us In debugging 
> 
> We can start using the javadoc plugin in pom.xml file
> 
> What do you think about this ?

Good idea. We can also add javadocs along with other changes when possible.

**Comment by ajaykathwate** [comment]
Hey @AmrElsayyad, Can I work on this ?
Can you explain this a bit more.

**Comment by AmrElsayyad** [comment]
> Hey @AmrElsayyad, Can I work on this ? Can you explain this a bit more.

Sure, go ahead!

**Comment by AmrElsayyad** [comment]
@ajaykathwate You need to add javadoc comments like this:

![image](https://github.com/user-attachments/assets/6def4355-77a5-441d-901a-f6f0d5435ca2)

Please let me know if you need any help.
